src/Bijectors.jl

```diff
@@ -335,16 +334,33 @@ function invlink(d::PDMatDistribution, Y::AbstractMatrix{<:Real})
     return LowerTriangular(X) * LowerTriangular(X)'
 end
 
-function logpdf_with_trans(d::PDMatDistribution, X::AbstractMatrix{<:Real}, transform::Bool)
-    lp = logpdf(d, X)
+function logpdf_with_trans(
+    d::PDMatDistribution, 
+    X::AbstractMatrix{<:Real}, 
+    transform::Bool
+)
+    T = eltype(X)
+    Xcf = cholesky(X, check=false)
+    if !issuccess(Xcf)
+        Xcf = cholesky(X + (eps(T) * norm(X)) * I)
+    end
+    lp = getlogp(d, Xcf, X)
     if transform && isfinite(lp)
-        U = cholesky(X).U
-        lp += sum((dim(d) .- (1:dim(d)) .+ 2) .* log.(view(U, diagind(U))))
-        lp += dim(d) * log(2)
+        U = Xcf.U
+        @inbounds @simd for i in 1:dim(d)
+            lp += (dim(d) - i + 2) * log(U[i, i])
```

Comment:
Could we please see some timings for this?

Reply:
```julia
using Distributions, LinearAlgebra, Bijectors, BenchmarkTools

function bench_tracker()
	d = Wishart(4, Matrix{Float64}(I, 4, 4))
	X = rand(d)
	@btime Bijectors.logpdf_with_trans($d, $X, true)
end

# No loop 
# 841.956 ns (11 allocations: 800 bytes)

# Loop
# 615.796 ns (6 allocations: 528 bytes)
```

Reply:
Sorry for the delay!

Reply:
```julia
function bench_tracker()
	d = InverseWishart(4, Matrix{Float64}(I, 4, 4))
	X = rand(d)
	@btime Bijectors.logpdf_with_trans($d, $X, true)
end

# No loop
# 881.271 ns (12 allocations: 1008 bytes)

# Loop
# 648.804 ns (7 allocations: 736 bytes)
```